PassKey

mantishK opened the pull request:
Initialize basex with a pass key for encrypted encoding. 
Usage - 
`baseX.Init(passKey)` 

The logic for encryption is as shown in [this link](http://kvz.io/blog/2009/06/10/create-short-ids-with-php-like-youtube-or-tinyurl/)

mantishK committed:
passward protection

mantishK committed:
comments and refactoring

mantishK commented:
Checks failing because I used `t.Run()` in tests (G0 1.7 Subtest feature), is there anything that can be done to update Go version of the CI or should I find alternate ways to test?

dineshappavoo commented:
@mantishK - Thanks for the pull request. You can update the travis config file to test with 1.7. Also it would be really great if you could give a description and use case for this PR. It will be helpful for others working on this package to understand your PR.

Sample travis.yml for 1.7:

```
language: go
go:
- 1.7
- tip
matrix:
  allow_failures:
      - go: tip
```

mantishK committed:
Changed go version for Travis

Go version for Travis needs to be 1.7 or more, because the tests include
subtest feature, which was added in 1.7

mantishK commented:
Since baseX is used to generate youtube like IDs, you may want the generated IDs to be secure. With the current implementation (on master), a particular number produces a particular Youtube like ID, this is easily reversible. A crawler can know what ID leads to what number. If it is encrypted with a password, the generated ID is dependent on the passkey. With different passkey, the ID for the same number is different. This way it is hard for anyone to guess what the number is, given the ID.
